### PR TITLE
feat(studio): expand sample agents, fix phishing agent

### PR DESCRIPTION
**Pull request by nv-odrulea**
## Summary

Generalizes Studio's "Create Example Agent" flow from a single hardcoded calculator config into a registry of canned examples, adds an **email phishing analyzer** example (tool code + config + eval), and wires matching eval configs into the Run Evaluation modal. Also fixes an agent-runtime bug that caused verbose, unparseable responses.

### New example agent: email phishing analyzer
- Vendors the NAT `email_phishing_analyzer` tool as an installed package at `plugins/nemo-agents/examples/email-phishing-analyzer/`, wired as a uv workspace member of `nemo-agents-plugin` so its custom `_type` resolves in the deploy runtime.
- Ships a curated agent config, eval config, and dataset.

### Studio: example sourcing refactor
- Replaces the inlined agent config in `const.ts` with a path-based registry at `src/constants/sampleAgents.ts`. Example agent/eval/dataset files now live as static assets under `web/packages/studio/public/sample-agents/<example>/` (fetched on demand, nothing bundled) — mirroring the existing `public/sample-datasets` pattern.
- New helpers `src/api/agents/{fetchSampleText,loadSampleAgentConfig}.ts`: fetch a public asset, parse the agent YAML, and inject the selected model.
- **Create Example Agent** modal: adds an example dropdown; fetches + parses + model-injects the config before creating.
- **Run Evaluation** modal: adds an example dropdown that **auto-matches** the selected agent to its eval config by name prefix; seeds the eval config + dataset from the static asset into the `{agent}-eval` fileset. `ensureEvalConfigFileset` gains an optional files arg (optimizer flow unchanged).

### Agent-runtime fix
- Phishing agent uses `tool_calling_agent` instead of `react_agent`. `react_agent` + `use_native_tool_calling` drops the ReAct stop-guard, so small models emitted and mis-parsed a runaway text scratchpad; `tool_calling_agent` uses native tool calls with no text format to mangle. Confirmed the model returns native `tool_calls` via the IGW.
- Forces a verdict-first final answer and hardens the eval judge prompt to extract the verdict from verbose output.

## Testing
- Studio: typecheck + lint clean; unit tests for the config loader and the agent→example matcher (incl. subset-prefix edge cases); updated create-flow test with a realistic mock handler; optimizer regression preserved.
- Plugin: `uv sync` installs the package; NAT registry resolves `_type: email_phishing_analyzer`.
- End-to-end: create → deploy → run evaluation verified against a local platform.

## Notes
- Studio keeps its own copy of the example data (independent of the plugin's CLI configs) so it can pin known-reliable models; the redundancy is intentional.
- Eval judge (`nvidia-nemotron-3-super-120b-a12b`) must exist as a VirtualModel in the target workspace.

## Summary by CodeRabbit

* **New Features**
  * Added new sample agents: an email phishing analyzer (tool-calling workflow + eval setup + test assets) and a calculator ReAct sample.
  * Enabled creating example agents by selecting from curated sample agents, with automatic config loading.
  * Default evaluation creation now seeds required eval files on demand (and avoids duplicate uploads).
* **Bug Fixes**
  * Improved sample-agent selection and model picker behavior for more reliable UI targeting.
* **Tests**
  * Added/updated unit tests and mocks covering sample-agent lookup, config loading, eval seeding, and parsing behavior.

### CODE REVIEW

File: web/packages/studio/src/routes/agents/AgentEvaluationsRoute/components/SubmitEvaluationModal.tsx [hunk 1]
```diff
@@ -205,14 +248,25 @@ export const SubmitEvaluationModal: FC<SubmitEvaluationModalProps> = ({
...
+            assetPath: example.evalConfigPath,
+            type: contentTypeForFile(example.evalConfigFileName),
+          },
+          {
+            path: example.evalDataFileName,
+            assetPath: example.evalDataPath,
+            type: contentTypeForFile(example.evalDataFileName),
+          },
+        ],
       };
```
[the remark this answers is not in the document]
Comment (nv-odrulea) [reply web/packages/studio/src/routes/agents/AgentEvaluationsRoute/components/SubmitEvaluationModal.tsx:279]: done

File: web/packages/studio/src/constants/sampleAgents.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+export const DEFAULT_SAMPLE_AGENT_KEY = SAMPLE_AGENTS[0].key;
+
+export const getSampleAgent = (key: string): SampleAgent =>
+  SAMPLE_AGENTS.find((agent) => agent.key === key) ?? SAMPLE_AGENTS[0];
+
+export const buildSampleAgentName = (namePrefix: string): string =>
+  `${namePrefix}-${Math.random().toString(36).slice(2, 8)}`;
+
+export const isSampleAgentName = (name: string): boolean =>
+  SAMPLE_AGENTS.some((agent) => name.startsWith(agent.namePrefix));
```
[the remark this answers is not in the document]
Comment (nv-odrulea) [reply web/packages/studio/src/constants/sampleAgents.ts]: done

File: web/packages/studio/src/api/agents/loadSampleAgentConfig.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  agentConfigPath: string,
+  modelName: string
+): Promise<Record<string, unknown>> => {
+  const text = await fetchSampleText(agentConfigPath);
+  const config = YAML.parse(text) as Record<string, unknown>;
+  const llms = config?.llms as { llm?: Record<string, unknown> } | undefined;
+  if (!llms?.llm) {
+    throw new Error(`Sample agent config ${agentConfigPath} is missing llms.llm`);
+  }
+  llms.llm.model_name = modelName;
```
[the remark this answers is not in the document]
Comment (nv-odrulea) [reply web/packages/studio/src/api/agents/loadSampleAgentConfig.ts]: done

File: plugins/nemo-agents/examples/email-phishing-analyzer/src/nat_email_phishing_analyzer/register.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+            # Handle missing or malformed fields with defaults
+            result = {
+                "is_likely_phishing": analysis.get("is_likely_phishing", False),
+                "explanation": analysis.get("explanation", "No detailed explanation provided"),
+            }
+
+            # Return as JSON string
+            return json.dumps(result)
+        except json.JSONDecodeError:
+            return "Error: Could not parse LLM response as JSON"
```
[the remark this answers is not in the document]
Comment (nv-odrulea) [reply plugins/nemo-agents/examples/email-phishing-analyzer/src/nat_email_phishing_analyzer/register.py:86]: done

File: web/packages/studio/public/sample-agents/email-phishing-analyzer/smaller_test.csv [hunk 5]
```diff
@@ -0,0 +1,25 @@
+subject,body,arrival_time,sender,intents,label,source,extra_info
```
Comment (htolentino-nvidia) [inline web/packages/studio/public/sample-agents/email-phishing-analyzer/smaller_test.csv:1]: There are 2 copies of this file on this PR, is that intentional?

**Review by htolentino-nvidia — APPROVED** [review]
LGTM, tested it deploying locally and it worked!

File: web/packages/studio/public/sample-agents/email-phishing-analyzer/smaller_test.csv (hunk 5 above)
Comment (nv-odrulea) [reply web/packages/studio/public/sample-agents/email-phishing-analyzer/smaller_test.csv:1]: yes, it is intentional.

the first copy, which lives in `/plugins/nemo-agents` is just a direct copy from the NAT repo /examples
it can be used as a reference for people coming from CLI
we need just the python part of it, because the tool type _email_phishing_analyzer must be registered as a python executable, so it can't live under studio's path.

on the other hand, when we create a new agent, and a new agent eval, we really just need some static config files.  Studio can't reach beyond its app boundaries into the plugins dir above to copy the files.  So what we have done so far, for calculator, is simply inline copy of the same configs.

I wanted to avoid that here, because the eval additionally needs some data rows in a csv file.

Duplicating ONLY the static configs in Studio's filepath gives us flexibility to modify these configs to known-good params (for example, models we know will work in build.nvidia.com provider) without being tied to canonical reference examples that might be more generic for CLI usage.  This allows us to tailor configs for working examples scoped specifically to Studio, while also making the original examples available higher up in the stack.

It's not my favorite pattern, but it's a reasonable compromise imo. Another option was to keep everything up in the `/plugins/nemo-agent` path, and have Vite copy over files at build time. That could maybe be a further optimization if we decided that Studio shoudl have the exact same config values as the global reference example?

Comment (nv-odrulea) [reply web/packages/studio/public/sample-agents/email-phishing-analyzer/smaller_test.csv:1]: when you create agent eval via API call, you don't POST the eval data file. you actually send a string reference to a fileset name.  the backend just looks inside that fileset for a config file, which in turn contains the path to the data file (if any).

so, in order to make an agent eval work, you need to seed a fileset with the correct config + data. the static copy that Studio has here is used to seed a unique fileset when a new agent-eval is created.
